text/dependencies/rfcs/0000-dependency-management-phase-two.md

```diff
@@ -0,0 +1,195 @@
+
+## Motivation
+
+Per top-level [RFC 0000:
+Overview](https://github.com/paketo-buildpacks/rfcs/blob/dependency-management-top-level/text/dependencies/rfcs/0000-dependency-management-overview.md),
+the second phase of the dependency management process update will be to
+completely rework the existing set of actions and workflows from the
+[dep-server](https://github.com/paketo-buildpacks/dep-server) such that the
+dep-server is no longer need because the entire process runs alongside the
+buildpacks.
+
+This phase will address concerns around the opaque complexity of the dep-server
+process. Additionally, it addresses the painpoint of maintainability.
+Dependencies maintainers will own the new workflows and actions, ensuring every
+relevant buildpack follows the same overall system architecture, while giving
+buildpacks maintainers freedom to make dependency-specific choices separately
+and debug issues with full authority.
+
+The new actions will also be simple to run locally in the event of
+Github Actions degradations, making the dependency-management system more
+flexible in use and transparent in process.
+
+## Detailed Explanation
+
+### Background
+For context, here is an enumeration of the current workflows that run in Github
+Actions within the dep-server repository.
+
+Each dependency has three main workflows:
+- [**Get New
+  Versions**](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/get-new-versions.yml):
+  discover new versions of the dependency by cross checking published versions
+  against a known-versions file on a timer and trigger `Build and Upload`
+  workflow when a new version is discovered.
+
+- [**Build and
+  Upload**](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/build-upload.yml):
+  each new dependency version is compiled or modified from it's upstream
+  source, uploaded to an S3 bucket, and the dependency metadata is gathered and
+  used to trigger the `Test and Upload` workflow.
+
+- [**Test and Upload
+  Metadata**](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/test-upload-metadata.yml):
+  runs tests against the compiled dependency and uploads dependency metadata to
+  S3 bucket if tests pass.
+
+Metadata and the dependencies themselves are stored in S3 buckets and are
+accessible through `https://api.deps.paketo.io/v1/dependency?name=<dependency>`
+through a series of API endpoint routing.
+
+Dependencies are updated in the buildpacks in a workflow that runs
+in the buildpack's Github Actions:
+- **Update Dependencies**: Runs every hour and runs [`jam
+  update-dependencies`](https://github.com/paketo-buildpacks/github-config/blob/f85a65d14fd90f6d63f1af2d408ee38f17ce5c0b/actions/dependency/update/action.yml#L45-L46)
+  against the `buildpack.toml`. If there are new versions available in the
+  dep-server, this workflow will discover them and open a PR to update the
+  `buildpack.toml`.
+
+### New Plan Overview
```

Comment:
I like this idea. @joshuatcasey and I are investigating what this looks like now and will update the RFC if we find it viable

Reply:
We may need to also pass in the OS/arch as arguments to the `make` command, because we've encountered a bug in which compilation code for bundler works differently for bionic vs jammy.

Reply:
We may also consider a `prepare` step that could do such things as install additional tools. It could be invoked as something like `make prepare os=<whatever>` so that it can handle the different operating systems.

Reply:
We may want to add a file such as `dependency/NAME` that contains the ID of the dependency.

Also, are we aware of any buildpacks that compile more than one ID at a time?

Reply:
We may also need a `prepare-pull-request` step that combines 6, 7, and 8